### PR TITLE
Change address constructor signature to const void* byte input 

### PR DESCRIPTION
**Pull request by Musteblume**
Address constructor change of unit32_t to void pointer. Added Tests for the constructor.

### CODE REVIEW

File: libvast/src/address.cpp [hunk 1]
```diff
@@ -21,6 +21,25 @@ address::address() {
...
 
+address::address(uint8_t const* bytes, family fam, byte_order order) {
+  auto b = reinterpret_cast<const uint32_t*>(bytes);
+  if (fam == ipv4) {
+    std::copy(v4_mapped_prefix.begin(), v4_mapped_prefix.end(), bytes_.begin());
+    auto p = reinterpret_cast<uint32_t*>(&bytes_[12]);
+    if (order == host)
+      *p = detail::to_network_order(*b);
+    else
+      *p = *b;
```
Comment (mavam) [inline libvast/src/address.cpp]: You can simplify this to this one-liner: `*p = order == host ? detail::to_network_order(*b) ? *b`.

File: libvast/src/address.cpp [hunk 2]
```diff
@@ -21,6 +21,25 @@ address::address() {
...
+  } else if (order == host) {
+    for (auto i = 0u; i < 4u; ++i) {
+      auto p = reinterpret_cast<uint32_t*>(&bytes_[i * 4]);
+      *p = detail::to_network_order(*(b + i));
+    }
+  } else {
+    std::copy(b, b + 4, reinterpret_cast<uint32_t*>(bytes_.data()));
```
Comment (mavam) [inline libvast/src/address.cpp]: More idiomatic: use `std::copy_n`.

File: libvast/test/address.cpp [hunk 3]
```diff
@@ -86,6 +94,10 @@ TEST(IPv6) {
   CHECK(f == (a ^ d));
   CHECK(f == e);
 
+  address g(raw8, address::ipv6, address::network);
```
Comment (mavam) [inline libvast/test/address.cpp]: Use brace initialization when possible. Here: `address g{...}`.

File: libvast/src/address.cpp (hunk 2 above)
Comment (mavam) [inline libvast/src/address.cpp]: Casting the input into 2 `uint64_t` operations cuts the operations in half.

(Probably won't matter in practice due to compiler optimizations, but still worthwhile pointing out).

**Comment by Musteblume** [comment]
Yes and done ;)

**Comment by mavam** [comment]
Minor detail for the future: [our style guide](https://github.com/vast-io/vast/blob/master/CONTRIBUTING.md#commit-messages) asks for no more than 50 characters in the first line of the commit message.

(This seemed too pedantic to warrant another iteration, so I went ahead and merged.)

**Pull request merged**